src/Components/MediaChannelComponent.ts

```diff
@@ -0,0 +1,176 @@
+ */
+export class MediaChannelComponent extends Component<MediaComponentSave> {
+
+    // MANDATORY: Define a name in ComponentNames.ts and place it here.
+    name: ComponentNames = ComponentNames.MEDIA_CHANNEL;
+    channelsArray: Array<TextChannel> = new Array<TextChannel>();
+    linkRegex: RegExp = /(https?:\/\/[^\s]+)/g; // not great but should work for all but weird edge cases
+    // may move to constants in future if needed?
+
+    async getSaveData(): Promise<MediaComponentSave> {
+        let ids = new Array<string>();
+        for (const c of this.channelsArray) {
+            ids.push((<TextChannel>c).id);
+        }
+        return {
+            channels: ids
+        };
+    }
+
+    async afterLoadJSON(loadedObject: MediaComponentSave | undefined): Promise<void> {
+        if (loadedObject) {
+            for (const c of loadedObject.channels) {
+                const channel = this.djmtGuild.getGuildChannel(c) as TextChannel;
+                if (!channel) {
+                    console.error(`[MediaChannelCheck]: could not load ${c}`);
+                    continue;
+                }
+                this.channelsArray.push(channel);
+            }
+        }
+    }
+
+    async onReady(): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    async onGuildMemberAdd(member: GuildMember): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    async onMessage(args: string[], message: Message): Promise<void> {
+        await this.checkMedia(message);
+    }
+
+    async onMessageReactionAdd(messageReaction: MessageReaction, user: User): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    async onMessageReactionRemove(messageReaction: MessageReaction, user: User): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    async onMessageUpdate(oldMessage: Message, newMessage: Message): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    async onMessageWithGuildPrefix(args: string[], message: Message): Promise<void> {
+        const command = args?.shift()?.toLowerCase() || '';
+        // Any interactive commands should be defined in ComponentCommands.ts
+        if (command === ComponentCommands.SET_MEDIA_CHANNEL) {
+            await this.setMediaChannel(args, message);
+        } else if (command === ComponentCommands.GET_MEDIA_CHANNEL) {
+            await this.getMediaChannel(message);
+        }
+        return Promise.resolve(undefined);
+    }
+
+    async onVoiceStateUpdate(oldState: VoiceState, newState: VoiceState): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    /**
+     * Toggles certain channels as "media channels", where any non-media messages will be deleted.
+     * @param args A list of channels to toggle.
+     * @param message The message object of the command message.
+     * @private
+     */
+    private async setMediaChannel(args: string[], message: Message): Promise<void> {
```

Comment:
I think it'd be really good to try to pull out some of the else statements in setMediaChannel here by using `return` more in the if statements.